pkg/color/coloring.go

```diff
@@ -89,6 +89,16 @@ func WrapIndices(s string, groups []int) string {
 	for i := 0; i < len(groups); i += 2 {
 		start := groups[i]
 		end := groups[i+1]
+
+		// check to see if the next group is an inner group
+		if i+2 < len(groups) {
```

Comment:
Would this handle double nested groups? eg..
`echo "1,2,3" | rare f -m "(((1)))"`

Reply:
It would (I checked);  it only stops panics from slice indexing like `groups[7:5]` by replacing them with `groups[5:5]`. 

However don't merge this,  a simple benchmark showed this takes 578 ns/op on my machine as opposed to 
`if start >= 0 && end >= 0 && end > start && start >= lastIndex`
as you had suggest, which takes only 138ns/op.